src/Control/Algebra.hs

```diff
@@ -79,51 +82,55 @@ type Has eff sig m = (Members eff sig, Algebra sig m)
 
 -- | Construct a request for an effect to be interpreted by some handler later on.
 send :: (Member eff sig, Algebra sig m) => eff m a -> m a
-send = alg . inj
+send = alg id . inj
 {-# INLINE send #-}
 
 
 -- base
 
 instance Algebra (Lift IO) IO where
-  alg (LiftWith with k) = with (Identity ()) coerce >>= k . runIdentity
+  alg hom (LiftWith with k) = with (Identity ()) (coerce . hom . runIdentity) >>= hom . k . runIdentity
 
 instance Algebra (Lift Identity) Identity where
-  alg (LiftWith with k) = with (Identity ()) coerce >>= k . runIdentity
+  alg hom (LiftWith with k) = with (Identity ()) (coerce . hom . runIdentity) >>= hom . k . runIdentity
 
 instance Algebra Choose NonEmpty where
-  alg (Choose m) = m True S.<> m False
+  alg hom (Choose m) = hom (m True) S.<> hom (m False)
```

Comment:
Terminal algebras only need to apply the homomorphism to any continuations or higher-order positions to produce an action in their carrier.